lib/engine.rb

```diff
@@ -8,19 +8,63 @@
 end
 
 module Engine
+  @games = {}
+
+  # Game or Meta
   GAMES = Game.constants.map do |c|
     klass = Game.const_get(c)
-    next if !klass.is_a?(Class) || klass == Game::Base
 
-    klass
+    if c.start_with?('G')
+      if klass.is_a?(Class)
+        klass
+      elsif klass.is_a?(Module)
+        klass::Meta
+      end
+    end
   end.compact
 
-  # Games that are alpha or above
+  # Game or Meta; all that are alpha or above
   VISIBLE_GAMES = GAMES.select { |game| %i[alpha beta production].include?(game::DEV_STAGE) }
 
+  # Game or Meta
   GAMES_BY_TITLE = GAMES.map { |game| [game.title, game] }.to_h
 
+  def self.get_game(game_meta)
+    require_tree "engine/game/#{game_meta.fs_name}" if RUBY_ENGINE == 'opal'
+
+    Engine::Game.constants
+      .map { |c| Engine::Game.const_get(c) }
+      .select { |c| c.constants.include?(:Game) }
+      .map { |c| c.const_get(:Game) }
+      .find { |c| c.title == game_meta.title }
+  end
+
+  # Game
+  def self.game_by_title(title)
+    return @games[title] if @games[title]
+    return @games[title] = GAMES_BY_TITLE[title] if GAMES_BY_TITLE[title].is_a?(Class)
+
+    game_meta = GAMES_BY_TITLE[title]
+    game = get_game(game_meta)
+
+    if game.nil? && RUBY_ENGINE == 'opal'
+      game_by_title(game_meta::DEPENDS_ON) if game_meta::DEPENDS_ON
+
+      src = "/assets/#{game_meta.fs_name}.js"
+
+      # add <script> tag to DOM to load the target game file
+      `var s = document.createElement('script');
+       s.type = 'text/javascript';
```

Comment:
could you instead do this with snabbdom?

Reply:
@tobymao I'm now working on getting this to work with snabbdom in the app layer, will re-open the PR when I've got it working.

Reply:
Moved to the app layer, but not with snabbdom.

The problem with snabbdom is that adding a script to the page will load it asynchronously, and to execute some code when it's done, the `onload` property can be used to define a script the browser calls when it's done. When this is done via a snabberb (ie `h(:script ...)`), `onload` doesn't get rendered to JS correctly.